src/core_plugins/kibana/public/dashboard/panel/panel_header/panel_options_menu.js

```diff
@@ -15,12 +18,11 @@ export class PanelOptionsMenu extends React.Component {
   toggleMenu = () => {
     this.setState({ isPopoverOpen: !this.state.isPopoverOpen });
   };
-
   closePopover = () => this.setState({ isPopoverOpen: false });
 
   onEditPanel = () => {
     this.closePopover();
-    this.props.onEditPanel();
+    window.location = this.props.editUrl;
```

Comment:
I wonder if setting `window.location` is the best way to set the URL -- I worry about the different ways we're setting the URL, e.g. `kbnUrl`, and the challenges we'll face when we unify them under a single routing system. If possible, I think it'd be best to be as consistent as possible, e.g. using `kbnUrl` everywhere. What do you think?

Reply:
Filed https://github.com/elastic/kibana/issues/14703.  since this was already here, I'd prefer to leave as if for this PR, but I agree it's a good point to bring up.  window.location is also being used in the visualize listing table.

The thing is, kbnUrl wasn't being used before, this is a replacement for an href link moving into an onClick.  Still not sure what the right answer is but agree something worthwhile to bring up, esp as more react gets introduced.